src/client.rs

```diff
@@ -2375,6 +2377,20 @@ mod test {
         assert_eq!(manifest.config.media_type, manifest::WASM_CONFIG_MEDIA_TYPE);
     }
 
+    #[tokio::test]
+    async fn test_pull_docker_ubuntu() {
+        let reference = Reference::try_from(DOCKER_IO_UBUNTU).expect("failed to parse reference");
```

Comment:
I'm afraid this test could report false negatives because of connectivity issues or rate limiting done by Docker Hub.

We are already starting a local instance of `distribution` during the execution of the unit tests. I think we should change this test to leverage this local instance, push something that uses a `OCI_IMAGE_INDEX_MEDIA_TYPE` inside of it and then try to pull it.

This would make the tests more robust

Reply:
Yes, agree, we also have many rate limits issue during use docker hub.

Will fix, thanks a lot!

Reply:
I found local registry: https://github.com/distribution/distribution don't support `application/vnd.oci.image.index.v1+json` even it is defined in : https://github.com/opencontainers/image-spec/blob/main/media-types.md#compatibility-matrix
it will have following error when we try to push the manifest:
```
ERRO[1363] response completed with error   err.code="manifest invalid" 
err.detail="mediaType in manifest list should be 'application/vnd.docker.distribution.manifest.list.v2+json' not 'application/vnd.oci.image.index.v1+json'" err.message="manifest invalid" go.version=go1.20 http.request.contenttype=application/vnd.docker.distribution.manifest.list.v2+json 
http.request.host="localhost:5005" http.request.id=331dff94-3942-47be-803a-c0c18c1b1e8d 
http.request.method=PUT http.request.remoteaddr="127.0.0.1:39880" http.request.uri=/v2/ubuntu/manifests/latest http.request.useragent= http.response.contenttype=application/json http.response.duration="126.053µs" http.response.status=400 http.response.written=82 instance.id=2b9ba786-7076-4a8a-a5f9-b87a7f720225 service=registry vars.name=ubuntu vars.reference=latest version=v2.7.0-2130-ge5d58108
```

Reply:
submitted the registry issue in: https://github.com/distribution/distribution/issues/3863